### PR TITLE
MOTOR-900 Fix synchro failures on PyMongo 4.1.1

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
All green including all syncho tests!

File: motor/core.py [hunk 1]
```diff
@@ -105,6 +105,8 @@ class AgnosticClient(AgnosticBaseProperties):
     arbiters = ReadOnlyProperty()
     close = DelegateMethod()
     __hash__ = DelegateMethod()
+    __next__ = DelegateMethod()
+    next = DelegateMethod()
```
Comment (ShaneHarvey) [inline motor/core.py:109]: Do we need these methods on Client/DB/Coll? I'd prefer _not_ to add the to motor unless we absolutely need to. Eventually the plan is to remove them from pymongo in 5.0 so I don't see a reason to add them to motor.

File: motor/core.py [hunk 2]
```diff
@@ -810,7 +836,7 @@ def __getitem__(self, name):
         )
 
         return collection_class(
-            self.database, self.name + "." + name, _delegate=self.delegate[name]
+            self.database, self.name + "." + str(name), _delegate=self.delegate[name]
```
Comment (ShaneHarvey) [inline motor/core.py:839]: I don't think we want to call str(name) since name should already be a str.

File: motor/motor_gridfs.py [hunk 3]
```diff
@@ -222,6 +222,7 @@ class AgnosticGridIn(object):
     readable = DelegateMethod()
     seekable = DelegateMethod()
     upload_date = ReadOnlyProperty()
+    md5 = ReadOnlyProperty()
```
Comment (ShaneHarvey) [inline motor/motor_gridfs.py:225]: Let's not add this property since it's useless in motor.

File: synchro/__init__.py [hunk 4]
```diff
@@ -483,7 +495,7 @@ def __getattr__(self, name):
 
     def __getitem__(self, name):
         # Access to collections with dotted names, like db.test['mike']
-        fullname = self.name + "." + name
+        fullname = self.name + "." + str(name)
```
Comment (ShaneHarvey) [inline synchro/__init__.py:498]: I don't think we want to call str(name) since name should already be a str.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: motor/core.py (hunk 1 above)
Comment (blink1073) [reply motor/core.py:109]: Sounds good, I should be able to skip the relevant tests instead

File: motor/core.py (hunk 2 above)
Comment (blink1073) [reply motor/core.py:839]: At least one of the tests was passing in an integer for the name

File: motor/motor_gridfs.py (hunk 3 above)
Comment (blink1073) [reply motor/motor_gridfs.py:225]: Ack, I'll find a way around this, might mean skipping a couple more tests.

File: synchro/__init__.py (hunk 4 above)
Comment (blink1073) [reply synchro/__init__.py:498]: Same comment as the core class

**Review by juliusgeo — APPROVED** [review]
LGTM!

File: motor/core.py (hunk 2 above)
Comment (ShaneHarvey) [reply motor/core.py:833]: We should skip those tests. Motor should not accept non-strings for getitem even if pymongo does.

File: synchro/__init__.py (hunk 4 above)
Comment (ShaneHarvey) [reply synchro/__init__.py:494]: We should skip those tests. Motor should not accept non-strings for getitem even if pymongo does.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: motor/core.py [hunk 5]
```diff
@@ -511,6 +518,7 @@ class AgnosticDatabase(AgnosticBaseProperties):
     __delegate_class__ = Database
 
     __hash__ = DelegateMethod()
+    __bool__ = DelegateMethod()
```
Comment (ShaneHarvey) [inline motor/core.py:521]: This implements the breaking change in PYTHON-808 for Motor 3.0, right? If so, could you make a note about this in the changelog?

Comment (blink1073) [reply motor/core.py:521]: Done

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Still waiting on the getitem str() issues.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**